### PR TITLE
feat(send): batching streams.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
@jrgm - I promised sending in batches by yesterday, sorry it's a day late!

I haven't tested any of this against a real SQS queue, it could be totally bunk.

https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/SQS.html#sendMessageBatch-property

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
WIP still, it's breaking.

**Comment by shane-tomlinson** [comment]
Looks like it's in better shape now, still untested against a real queue.

File: bin/json-to-sqs.js [hunk 1]
```diff
@@ -41,23 +41,56 @@ const reader = new JSONReader({
...
   let sqsWriter;
+  let sqsMock;
   if (program.dryrun) {
-    sqsWriter = new SQSWriter({
-      sqs: {
-        sendMessage(message, callback) {
-          setTimeout(() => callback(), Math.random() * 5);
-        }
+    sqsMock = {
+      sendMessage(message, callback) {
```
Comment (jrgm) [inline bin/json-to-sqs.js]: Nit: it might be less clutter if this was moved into a `createSqsMock()` helper function.

**Comment by jrgm** [comment]
>  I made the changes to accept option `--highwater N`
... and pass `highWaterMark: N` to the streams' constructors.

File: bin/regenerate-testdata.sh [hunk 2]
```diff
@@ -6,12 +6,17 @@ SALESFORCE=./test_data/salesforce.csv
 SALESFORCE_SORTED=./test_data/salesforce_sorted.csv
 EXPECTED_COUNTS=./test_data/expected.json
 
-COUNT=4000
+COUNT=50000
```
Comment (shane-tomlinson) [inline bin/regenerate-testdata.sh]: revert this to 4000.

**Comment by shane-tomlinson** [comment]
@jrgm - mind another review?

**Comment by jrgm** [comment]
r+

**Pull request merged**